trading_calendars/exchange_calendar_xwbo.py

```diff
@@ -57,7 +62,13 @@
 
 SaintStephensDay = Holiday("Saint Stephen's Day", month=12, day=26)
 
-NewYearsEve = new_years_eve()
+# Prior to 2016, when New Year's Eve fell on the weekend, it was observed
+# on the preceding Friday. In 2016 and after, it is not made up.
+NewYearsEveThrough2015 = new_years_eve(
+  observance=previous_friday,
+  end_date='2016',
+)
+NewYearsEve = new_years_eve(start_date='2016')
```

Comment:
I've called other holidays like this "NewYearsEveOnOrAfter2016" to avoid possible confusion where someone uses this without realizing that it has a start date.